spec/acceptance/suites/rpm_docker/nodesets/el7.yml

```diff
@@ -71,11 +71,35 @@ HOSTS:
       - 'yum install -y ntpdate rubygems'
 
       # RVM
-      - 'runuser build_user -l -c "{ gpg2 --keyserver hkp://pgp.mit.edu --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 || gpg2 --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3; } || { gpg2 --keyserver hkp://pgp.mit.edu --recv-keys 7D2BAF1CF37B13E2069D6956105BD0E739499BDB || gpg2 --keyserver hkp://keys.gnupg.net --recv-keys 7D2BAF1CF37B13E2069D6956105BD0E739499BDB ; }"'
-      - 'runuser build_user -l -c "curl -sSL https://get.rvm.io | bash -s stable"'
+      #
+      # Do our best to get one of the keys from at one of the servers, and to
+      # trust the right ones if the GPG keyservers return bad keys
+      #
+      # These are the keys we want:
+      #
+      #  409B6B1796C275462A1703113804BB82D39DC0E3 # mpapis@gmail.com
+      #  7D2BAF1CF37B13E2069D6956105BD0E739499BDB # piotr.kuczynski@gmail.com
+      #
+      # See:
+      #   - https://rvm.io/rvm/security
+      #   - https://github.com/rvm/rvm/blob/master/docs/gpg.md
+      #   - https://github.com/rvm/rvm/issues/4449
+      #   - https://github.com/rvm/rvm/issues/4250
+      #   - https://seclists.org/oss-sec/2018/q3/174
+      #
+      # NOTE (mostly to self): In addition to RVM's documented procedures,
+      # importing from https://keybase.io/mpapis may be a practical
+      # alternative for 409B6B1796C275462A1703113804BB82D39DC0E3:
+      #
+      #    curl https://keybase.io/mpapis/pgp_keys.asc | gpg2 --import
+      #
+      - "runuser build_user -l -c '{ gpg2 --keyserver hkp://pgp.mit.edu --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 || gpg2 --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3; }; key1=$?; { gpg2 --keyserver hkp://pgp.mit.edu --recv-keys 7D2BAF1CF37B13E2069D6956105BD0E739499BDB || gpg2 --keyserver hkp://keys.gnupg.net --recv-keys 7D2BAF1CF37B13E2069D6956105BD0E739499BDB ; }; key2=$? ; [ \"$key1\" -eq 0 ] || [ \"$key2\" -eq 0 ] || curl -sSL https://rvm.io/mpapis.asc | gpg --import -'"
```

Comment:
What value does rvm provide in this situation? I would try installing ruby from source at this point.

Reply:
@jeefberkey OS specific patches and package installation that we don't have to worry about on a case-by-case basis as well as validating our development environment instructions in simp-doc